logger/utils.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package logger
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/aws/shim-loggers-for-containerd/debug"
+)
+
+func retryWithExponentialBackoff(operation func() error, initialBackoff time.Duration, maxBackoff time.Duration) error {
+	attempt := 0
+	reachMaxBackoff := false
+	for {
+		err := operation()
+		if err == nil {
```

Comment:
This is a pretty weird line of code to read at first glance. Maybe we can add a comment like `// only exit if there's no error`?

Reply:
if we keep increasing attempt, the value of backoff below may exceed the int limit
> backoff := float64(initialBackoff) * math.Pow(2, float64(attempt))